common/src/main/java/io/netty5/util/concurrent/AbstractScheduledEventExecutor.java

```diff
@@ -15,33 +15,28 @@
  */
 package io.netty5.util.concurrent;
 
-import io.netty5.util.internal.DefaultPriorityQueue;
-import io.netty5.util.internal.PriorityQueue;
 import io.netty5.util.internal.PriorityQueueNode;
 
-import java.util.Comparator;
-import java.util.Queue;
 import java.util.concurrent.Callable;
 import java.util.concurrent.TimeUnit;
 
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.Executors.callable;
 
 /**
  * Abstract base class for {@link EventExecutor}s that want to support scheduling.
  */
 public abstract class AbstractScheduledEventExecutor extends AbstractEventExecutor {
 
-    private static final Comparator<RunnableScheduledFutureNode<?>> SCHEDULED_FUTURE_TASK_COMPARATOR =
-            Comparable::compareTo;
-    private static final RunnableScheduledFutureNode<?>[]
-            EMPTY_RUNNABLE_SCHEDULED_FUTURE_NODES = new RunnableScheduledFutureNode<?>[0];
-
-    private PriorityQueue<RunnableScheduledFutureNode<?>> scheduledTaskQueue;
+    private TaskScheduler taskScheduler;
+    private TaskSchedulerFactory taskSchedulerFactory;
```

Comment:
**Scalability Issue**: The introduction of TaskScheduler and TaskSchedulerFactory as fields increases the memory footprint of each AbstractScheduledEventExecutor instance. In a highly scalable system, where potentially millions of these executors could be instantiated, this additional memory usage could become significant. Moreover, the dynamic selection of TaskScheduler based on a factory pattern, while providing flexibility, introduces additional overhead during executor instantiation, which could impact the startup time of systems with a large number of executors. <br> **Fix**: Consider using a static, shared TaskScheduler instance for common scheduling scenarios, reducing the memory footprint per executor instance. This shared instance could be a highly efficient, multi-tenant scheduler designed to handle tasks from multiple executors concurrently. For use cases requiring specialized scheduling, continue to allow the injection of a custom TaskScheduler, but ensure this is an exceptional scenario rather than the norm. <br> **Code Suggestion**: 
 ```
 -    private TaskScheduler taskScheduler;
 -    private TaskSchedulerFactory taskSchedulerFactory;
 +    private static final TaskScheduler sharedTaskScheduler = DefaultTaskSchedulerFactory.INSTANCE.newTaskScheduler(this);

      protected AbstractScheduledEventExecutor(TaskSchedulerFactory taskSchedulerFactory) {
 -        this.taskSchedulerFactory = requireNonNull(taskSchedulerFactory, "taskSchedulerFactory");
 +        // Use sharedTaskScheduler for common cases
      }

 +    TaskScheduler taskScheduler() {
 +        return sharedTaskScheduler;
 +    }
 ```